wazp/pages/04_dashboard.py

```diff
@@ -1,15 +1,228 @@
+                                    "likelihood",
+                                    style={
+                                        "width": "15%",
+                                        "float": "right",
+                                        "margin-top": "110px",
+                                    },
+                                    labelStyle={
+                                        "display": "block",
+                                        "fontSize": ".8rem",
+                                    },
+                                ),
+                            ],
+                            style={"width": "49%", "display": "inline-block"},
+                        ),
+                        html.Div(
+                            [
+                                html.Div(
+                                    [
+                                        dcc.Input(
+                                            id="bin-size",
+                                            type="number",
+                                            placeholder="nbins per axis",
+                                            debounce=True,  # need to enter,
+                                        )  # "value" will be in callback
+                                    ],
+                                    style={
+                                        "float": "right",
+                                    },
+                                ),
+                                dcc.Graph(  # figure first row right
+                                    id="graph-heatmap",
+                                    figure=fig_heatmap,
+                                    style={
+                                        "width": "85%",
+                                        "display": "inline-block",
+                                        "float": "right",
+                                    },
+                                ),
+                            ],
+                            style={"width": "49%", "display": "inline-block"},
+                        ),
+                    ]
+                ),
+                html.Div(  # second row figures
+                    children=[
+                        dcc.Graph(
+                            id="graph-barplot",
+                            figure=fig_barplot,
+                            style={"width": "42%", "display": "inline-block"},
+                        ),
+                        dcc.Graph(
+                            id="graph-entries-exits",
+                            figure=fig_entries_exits,
+                            style={
+                                "width": "42%",
+                                "display": "inline-block",
+                                "float": "right",
+                            },
+                        ),
+                    ]
+                ),
+            ]
```

Comment:
Might be more readable to build these guys up as local-scope variables.

What about a bunch of functions:
```python

def first_row_left():
    """Configure the HTML div contents for the first row left hand side"""

    graph = dcc.Graph(
       id="graph-trajectories",
       figure=fig_trajectories,
       style={
          "width": "85%",
          "display": "inline-block",
       }
    )
    # ...
    return html.Div(children=[graph, ...], ...)

    

layout = html.Div(
    className="row",
    children=[
        html.H1(children="Dashboard"),
        html.H1("Dashboard"),
        html.Br(),
        html.H5(
            "Input data", style={"margin-top": "20px", "margin-bottom": "10px"}
        ),
        html.Div(children=[], id="table-container"),
        html.Div([first_row_left(), first_row_right() ...])
        # ...
```
...?

Reply:
The functions might be a bit [WET](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself#WET). If we add some tests, we can refactor.